provisioner/ansible/provisioner_test.go

```diff
@@ -720,6 +728,7 @@ func TestUseProxy(t *testing.T) {
 			Reader: new(bytes.Buffer),
 			Writer: new(bytes.Buffer),
 		}
+		//nolint:errcheck
 		p.Provision(ctx, ui, comm, tc.generatedData)
```

Comment:
When adding an err check here I found that the test was failing because the mock is returning an error for not using the happyPath. It is not clear to me what the expected behavior is for this test case number 1. 

```
2021/04/13 15:06:32 ui: Provisioning with Ansible...
--- FAIL: TestUseProxy (0.04s)
    provisioner_test.go:733: use_proxy is true; we should set up adapter failed to provision with chose sadpath
FAIL
FAIL 
```

Reply:
maybe the mock has some weird logic to be changed according to inputs 🤔  this is odd.